en/docs/develop/api-policies/create-a-policy.md

```diff
@@ -0,0 +1,291 @@
+
+
+# Sets the given `mediation:ResourcePath` instance as the resource path of this context.
+public function setResourcePath(ResourcePath path);
+
+
+# Adds a mapping between a path param name and a resolved value for it. There need not be a path parameter in the
+# resource path by the name specified in `name` for one to use this method. On its own, the path param values have
+# no bearing on the resource path.
+public function addPathParamValue(string name, PathParamValue value);
+
+
+# Returns the collection of resolved values for the path parameters in this particular context, mapped
+# by the parameter name.
+public function resolvedPathParams() returns map<PathParamValue> & readonly;
+
+
+# Removes the resolved path parameter value which maps to the specified name.
+public function removePathParamValue(string name);
+
+
+# Adds a query parameter to the request to be sent to the backend. If there is already a query parameter by
+# with the same name, the new value will be appended to it, making it an array.
+public function addQueryParam(string name, string value);
+
+
+# Removes the specified query parameter from the request. If the value of the parameter is an array, the whole
+# array will be removed.
+public function removeQueryParam(string name);
+
+
+# Retrieves a map of all the query params in the current request context. The returned map is a read only snapshot
+# of the map of query parameters in the context at the time this method was called.
+public function queryParams() returns map<string[]> & readonly;
+```
+
+### Keywords
+
+The following key words are used in the Ballerina.toml file for mediation policies to work:
+
+- **choreo-apim-mediation-policy**: Mandatory keyword which identifies this package is a mediation policy type.
+- **choreo-apim-mediation-request-flow**: Specifies whether the policy applies to the request flow.
+- **choreo-apim-mediation-response-flow**: Specifies whether the policy applies to the response flow.
+- **choreo-apim-mediation-fault-flow**: Specifies whether the policy is applicable for the fault flow.
+
+### Policy name and description
+
+The Package.md file contains information about the policy. Choreo uses this information to render the policy configuring UI. This file is written in Markdown format and should be structured as follows.
+
+**Format:**
+```
+# <policy-name>
+
+
+## Overview
+
+
+<policy description>
+
+```
+
+**Example:**
+```
+# ValidateHeader
+
+
+## Overview
+
+
+This policy validates the request and response headers with the configured values.
+
+```
+
+### Policy versioning
+
+Choreo or mediation dependencies bump their major version when switching to a major update in the language. For example, from update 1 to update 2. There can be significant incompatibilities in these major versions. Therefore, we recommended versioning the policy package in a manner that the major version gets bumped when upgrading the Choreo/mediation dependency version to a major version.  
```

Comment:
Suggestion:
"Choreo or mediation dependencies bump their major version when switching to a major update in the language (for example, from update 1 to update 2). There can be significant incompatibilities in these major versions. Therefore, we recommended versioning the policy package in a manner where the major version gets bumped when upgrading the Choreo/mediation dependency version to a major version."

Reply:
done